pkg/process/monitor/process_monitor.go

```diff
@@ -0,0 +1,281 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package monitor
```

Comment:
i do miss UTs for the package
not everything can be easily tested
but some logic can be

I tested couple of functionalities
```go
func TestProcessMonitor(t *testing.T) {
	// Making sure we get the same process monitor if we call it twice.
	pm, err := GetProcessMonitor()
	require.NoError(t, err)

	pm2, err := GetProcessMonitor()
	require.NoError(t, err)

	require.Equal(t, pm, pm2)

	// Sanity subscribing a callback.
	callback := &ProcessCallback{
		Event:    EXEC,
		Metadata: ANY,
		Callback: func(pid uint32) {},
	}
	unsubscribe, err := pm.Subscribe(callback)
	require.NoError(t, err)

	// duplicated subscription should fail.
	_, err = pm.Subscribe(callback)
	require.Error(t, err)

	// making sure unsubscribe works and does not panic for the second unsubscription.
	unsubscribe()
	require.NotPanics(t, unsubscribe)

	numberOfExecs := 0
	numberOfExits := 0

	tmpFile, err := ioutil.TempFile("", "echo")
	require.NoError(t, err)
	defer os.Remove(tmpFile.Name())
	_, err = fileutils.CopyFile("/usr/bin/echo", tmpFile.Name())
	require.NoError(t, err)

	require.NoError(t, os.Chmod(tmpFile.Name(), 0500))

	require.NoError(t, pm.Initialize())
	// Sanity subscribing a callback.
	callbackExec := &ProcessCallback{
		Event:    EXEC,
		Metadata: NAME,
		Regex:    regexp.MustCompile(path.Base(tmpFile.Name())),
		Callback: func(pid uint32) {
			numberOfExecs++
		},
	}
	callbackExit := &ProcessCallback{
		Event:    EXIT,
		Metadata: NAME,
		Regex:    regexp.MustCompile(path.Base(tmpFile.Name())),
		Callback: func(pid uint32) {
			numberOfExits++
		},
	}

	unsubscribeExec, err := pm.Subscribe(callbackExec)
	require.NoError(t, err)
	unsubscribeExit, err := pm.Subscribe(callbackExit)
	require.NoError(t, err)

	require.NoError(t, exec.Command(tmpFile.Name(), "test").Run())
	require.Eventuallyf(t, func() bool {
		return numberOfExecs == 1 && numberOfExits == 1
	}, time.Second, time.Millisecond*200, "didn't capture exec and exit")

	unsubscribeExec()
	require.NoError(t, exec.Command(tmpFile.Name()).Run())
	require.Eventuallyf(t, func() bool {
		return numberOfExecs == 1 && numberOfExits == 2
	}, time.Second, time.Millisecond*200, "didn't capture exec and exit")

	unsubscribeExit()
	require.NoError(t, exec.Command(tmpFile.Name()).Run())
	require.Eventuallyf(t, func() bool {
		return numberOfExecs == 1 && numberOfExits == 2
	}, time.Second, time.Millisecond*200, "didn't capture exec and exit")
}

```

Reply:
Thanks for the unit test
This will not work as exit process don't evaluate the filter as the process is gone
But that another good point of view of this monitor

Reply:
the test did work locally, including the exit

Reply:
not if you are under process load, your will count all process exits (not only the one you launch)
I changed the behavior, depending if you register the exit callback with ANY or something else
I updated and documented the exit callbacks behaviour